### DISCUSSION TITLE
Configuring and working with passive/unactuated joints (joints that are not considered for motion planning)

### DISCUSSION THREAD

**Discussion by jsupratman13** [NVlabs/curobo#90] [Software Q&A] [answered]
Hi

I'm working with a robot arm that includes a linear unit. I'd like cuRobo to motion plan for the robot arm but not for the linear unit. Can I manually update the linear unit's position in cuRobo so it can adjust its joint state accordingly for collision avoidance?
I've come across the `lock_joints` parameter, but I was wondering if it is possible to adjust the joint position and have cuRobo recognize these changes for accurate collision avoidance and planning. If so, could you guide me on how to use `lock_joints` to update these joint states in cuRobo for accurate collision avoidance planning? Any insights or instructions would be greatly appreciated.

**Comment by balakumar-s** [comment]
This is not currently exposed as an easy to use API.

We can look into this in two weeks.

How often do you want to be able to change the state of locked joints? There will be a 0.1 second overhead as we have to recompute kinematic parameters every time the locked state changes.

**Reply by jsupratman13** [reply]
Thanks for the response and the willingness to explore this feature.

Regarding the frequency of changes to the locked joints, I plan to update them only before generating a motion plan. Updates will be occasional and specific to the planning stage, rather than continuous or frequent during operation. The overhead for recomputing kinematic parameters seems manageable for my application if the overhead is a one-time cost per planning session.

**Reply by balakumar-s** [reply, marked as answer]
This feature is now available: https://github.com/NVlabs/curobo/blob/f25281e9302ba5bcc7860c3019a7073876d8b89d/tests/motion_gen_api_test.py#L64
